ocean/src/clans.rs

```diff
@@ -1,39 +1,57 @@
-
+use std::cell::RefCell;
+use std::collections::HashMap;
 #[derive(Debug)]
 pub struct ClanSystem {
-    // TODO: add necessary fields
+    clans: RefCell<HashMap<String, Vec<String>>>,
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem{
+            clans: RefCell::new(HashMap::new())
+        }
     }
 
     /**
      * Returns a list of the names of the clan members for the given clan id.
      */
     pub fn get_clan_member_names(&self, clan_id: &str) -> Vec<String> {
-        unimplemented!();
+        let clan = self.clans.borrow().get(clan_id).unwrap_or(&Vec::new()).clone();
+        clan
     }
 
     /**
      * Returns the number of clans currently in existence.
      */
     pub fn get_clan_count(&self) -> usize {
-        unimplemented!();
+        self.clans.borrow().len()
     }
 
     /**
      * Returns the number of clan members for the given clan id.
      */
     pub fn get_clan_member_count(&self, clan_id: &str) -> usize {
-        unimplemented!();
```

Comment:
same as above - we can just return statement instead of assigning it to a variable